enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,236 @@
+  - TBD
+creation-date: 2020-07-16
+last-updated: 2020-07-16
+status: implementable
+see-also:
+  - "/enhancements/installer/connected-assisted-installer.md"  
+---
+
+# Assisted Installer On-Prem
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The assisted installer, [previously proposed and
+accepted](connected-assisted-installer.md), includes a service running on
+cloud.redhat.com to which agents connect and wait for orders on how to
+A) provision the host they are running on, and B) join or form a cluster.
+
+This enhancement proposes to enable a user to run that "assistant" service
+on-premise, inside their own network and on their own infrastructure, for
+one-time-use to create a new cluster
+
+## Motivation
+
+Many users will not want a service that can provision hosts inside their
+network to be run by a third party outside of their network. Reasons can
+include trust, control, and reproducibility. Thus those users will prefer to
+run the service on-premise.
+
+### Goals
+
+* Users can run the entire assisted install workflow with software that runs inside their network.
+
+### Non-Goals
+
+* Disconnected and air-gapped environments are not included in this enhancement.
+* Adding Nodes on day 2 will be addressed separately.
+* Utilizing a local container registry for mirrored content will be addressed in a future enhancement.
+
+## Proposal
+
+A user will visit cloud.redhat.com to download a live ISO with which they can
+run the assisted installer on-premise. The RHCOS-based ISO will include:
+
+* the user's pull secret
+* a systemd unit file that runs all containers necessary to run the assisted installer
+
+Similar to downloading the current installer, the download will default to the
+latest available version of the assisted installer, RHCOS and OpenShift.
+
+### User Stories [optional]
+
+#### Story 1
+
+I want to install OpenShift while running the assisted installer and the entire
+installation workflow on infrastructure that I control. I do not want a third
+party cloud service to control the provisioning of my hosts from across the
+internet.
+
+#### Story 2
+
+I want to automate use of the assisted installer and expect reproducible
+results. Utilizing a cloud service does not provide sufficient control or
+guarantees of reproducibility.
+
+#### Story 3
+
+I want to test the assisted installer prior to production use and expect
+reproducible results. Utilizing a cloud service does not provide sufficient
+control or guarantees of reproducibility.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Single Use on Day 1 Only
+
+The on-premise assisted installer will run once per cluster. As a live ISO, it
```

Comment:
Do we need to generate a new ISO for each version of OpenShift that is released?

Reply:
I think we can just use the RHCOS [image that's normally used for UPI](https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.4/latest/). The one for OCP 4.4 hasn't changed since May 5, so I'm guessing we won't need to worry about updating too often. However we will need to be on the current Y release and may want to coordinate with the OCP release schedule.